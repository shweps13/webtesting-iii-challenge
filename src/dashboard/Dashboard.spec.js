// Test away
import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import renderer from "react-test-renderer";

import Dashboard from './Dashboard';


test('Dashboard renders without crashing', () => {
  render(<Dashboard />);
});

test('Matches snapshot', () => {
    const tree = renderer.create(<Dashboard />); 
    expect(tree.toJSON()).toMatchSnapshot();
  });

test('Display element showings on Dashboard page', () => {

    const { getByTestId } = render(<Dashboard />);
    getByTestId(/display-element/i);
  });
 
test('Controls element showings on Dashboard page', () => {

    const { getByTestId } = render(<Dashboard />);
    getByTestId(/controls-element/i);
});

test('Showing [open/close] when click on it', () => {

    const { getByTestId, queryByText } = render(<Dashboard />);
    expect(queryByText(/Open/i)).toBeTruthy();
    fireEvent.click(getByTestId(/control-btn2/i));
    expect(queryByText(/Closed/i)).toBeTruthy();
});