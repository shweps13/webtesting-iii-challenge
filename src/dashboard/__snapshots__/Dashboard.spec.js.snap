// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches snapshot 1`] = `
Array [
  <div
    className="display panel"
    data-testid="display-element"
  >
    <div
      className="led green-led"
    >
      Unlocked
    </div>
    <div
      className="led green-led"
    >
      Open
    </div>
  </div>,
  <div
    className="controls panel"
    data-testid="controls-element"
  >
    <button
      className="toggle-btn"
      data-testid="control-btn1"
      disabled={true}
      onClick={[Function]}
    >
      Lock Gate
    </button>
    <button
      className="toggle-btn"
      data-testid="control-btn2"
      disabled={false}
      onClick={[Function]}
    >
      Close Gate
    </button>
  </div>,
]
`;
